test/rhsmlib/dbus/test_register.py

```diff
@@ -315,6 +320,17 @@
 class RegisterDBusObjectTest(DBusServerStubProvider):
     dbus_class = RegisterDBusObject
     dbus_class_kwargs = {}
+    socket_dir: Optional[tempfile.TemporaryDirectory] = None
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.socket_dir = tempfile.TemporaryDirectory()
+        cls.addClassCleanup(cls.socket_dir.cleanup)
+
+        socket_dir_patch = mock.patch.object(DomainSocketServer, "_server_socket_path", cls.socket_dir.name)
+        socket_dir_patch.start()
+
+        super().setUpClass()
```

Comment:
I can see failed CI tests (temporary file was deleted for some reason). When I change this to `setUp()`, then it works locally:

```python
    def setUp(self) -> None:
        self.socket_dir = tempfile.TemporaryDirectory()
        self.addCleanup(self.socket_dir.cleanup)

        socket_dir_patch = mock.patch.object(DomainSocketServer, "_server_socket_path", self.socket_dir.name)
        socket_dir_patch.start()
```

Maybe, we can try this approach and it will be more reliable.